server_modules/challenges.js

```diff
@@ -0,0 +1,16 @@
+var utils = require('./utils');
+var fileSystem = require('./file_system');
+
+exports.route = function(app) {
+
+    //get challenges over xhr, for hosted service
+    app.get('/challenge/:year', function(req, res) {
+        var file = '/challenges/js/{0}.js'.format(req.params.year);
+
+        fileSystem.readFile(file).then(function(data) {
```

Comment:
`readFile()` will call `absolutePath`, but because the filename is already absolute, it will try to use `/challenges/...` as-is (i.e. no root).